Modules/_testcapimodule.c

```diff
@@ -2340,6 +2340,177 @@ get_timezone_utc_capi(PyObject* self, PyObject *args) {
     }
 }
 
+static PyObject *
+get_date_fromdate(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int year = 0, month = 0, day = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(args, "iii|p", &year, &month, &day, &macro)) {
```

Comment:
I would prefer to make macro mandatory. Maybe it should be moved at the beginning, but I would prefer to get Paul's call on that.